### DISCUSSION TITLE
Autostart works with Systemd, but server stop sending and receiving mails

### DISCUSSION THREAD

**Discussion by impirecom** [postalserver/postal#2556] [Help with using Postal]
Hello,

I followed guides like this one:

https://community.blynk.cc/t/howto-use-systemd-instead-of-rc-local-on-linux-to-auto-start-the-server/20582

An implemented the following:

#cd /etc/systemd/system
#nano postal.service

[Unit]
Description=Postal Service
Requires=docker.service
After=docker.service

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/bin/postal start
ExecStop=/usr/bin/postal stop
TimeoutStartSec=0

[Install]
WantedBy=multi-user.target

#systemctl preset postal.service
#service postal start

System booted and server seems fine, but all mails stopped sending and receiving. Everything go to queue. I restored back to the back up that doesn't have this autoboot setup, and server works fine again.

Any idea what is causing this?

Thank you in advance for your help.

Kind regards,

Andy

**Comment by willpower232** [comment]
If you're using the official install method with Docker then you probably just need to make this modification rather than using systemd https://github.com/postalserver/install/pull/3

If you're hosting the database and rabbitmq on the same server, we have seen those take a little while to get started which means the attempt to start postal fails because those other two aren't ready and waiting yet which is another reason to include restart directives in the docker-compose.yml file.

**Reply by impirecom** [reply]
Thank you @willpower232 

I do have the database and rabbitmq on the same server. What would be the proper steps to get the autostart to work? The link you've sent, it seems folks still haven't found the solution.

**Reply by willpower232** [reply]
If you have them as docker containers, you would need to specify the "restart unless stopped" thing for them.

If they are traditionally installed, then you probably just need to `sudo systemctl enable <service>`

**Reply by impirecom** [reply]
I have them as docker containers. Can you please share the exact steps for auto start postal?

**Reply by willpower232** [reply]
You need to make this change to your docker-compose file https://github.com/postalserver/install/pull/3/files

**Reply by impirecom** [reply]
Thank you. That's it, just copy the entire file to replace the current? Anything else needed in order for Postal to autostart upon boot?

**Reply by willpower232** [reply]
Or just add the changes from the PR, your call. You will probably need to `postal stop` and `postal start` to make sure the changes apply.

**Reply by impirecom** [reply]
Very sorry. Don't understand what it mean by "just add the changes from the PR"

**Reply by impirecom** [reply]
Just added "restart: unless-stopped" to each section of the file, exactly like the file in https://github.com/postalserver/install/pull/3/files. Rebooted. Postal still require a manual start with "postal start command". What did I do wrong?

**Reply by willpower232** [reply]
> You will probably need to `postal stop` and `postal start` to make sure the changes apply.

**Reply by impirecom** [reply]
Okay. that worked really well. Thank you so much!
